includes/wc-core-functions.php

```diff
@@ -2042,6 +2042,7 @@ function wc_enable_wc_plugin_headers( $headers ) {
 }
 add_filter( 'extra_theme_headers', 'wc_enable_wc_plugin_headers' );
 add_filter( 'extra_plugin_headers', 'wc_enable_wc_plugin_headers' );
+wp_cache_delete( 'plugins', 'plugins' );
```

Comment:
We can't leave this code in here, it will delete the cache for plugins in all page loads in admin and frontend.
This is bad for performance.

Reply:
This cache is only used in the `get_plugins()` function, which is defined in `wp-admin/includes/plugin.php`. So it won't affect frontend requests.

For admin requests, there is a performance hit, but only if there's a poorly coded plugin installed alongside of WooCommerce. In that case (a plugin that manages to call `get_plugins()` before `woocommerce/woocommerce.php` is even run), there's absolutely no other way of fixing this issue.

Except maybe renaming `WooCommerce` to `AaaCommerce` :)

Reply:
@DanReyLop this file runs in all pages, this is not a good place for this code.

Reply:
Calling `wp_cache_delete` on a non-existent key is a no-op. Would it be better to call it in the [`WC_Admin` constructor](https://github.com/woocommerce/woocommerce/blob/4f485a14deaedff4ba41fb95a1d50b180108563e/includes/admin/class-wc-admin.php#L22) anyway?

As an alternative, in https://github.com/woocommerce/woocommerce/pull/25237/commits/6864533f7e2fb89428113310465740ff3f12fccc I clear the cache only before it's needed. The extra plugin headers are needed in only 2 scenarios:
- On the pages related with plugin updates (the `WC tested up to` header is used)
- On the WC Helper pages and notifications (the `Woo` header is used)